src/Middleware/Diagnostics/src/ExceptionHandler/ExceptionHandlerMiddlewareImpl.cs

```diff
@@ -180,27 +178,11 @@ private async Task HandleException(HttpContext context, ExceptionDispatchInfo ed
                 }
             }
 
-            if (!handled)
+            if (!handled && _options.ExceptionHandler is not null)
```

Comment:
I feel like this is a breaking change in behavior. Assuming `_problemDetailsService` is not null, and [`AddProblemDetails`](https://github.com/dotnet/aspnetcore/blob/933bc8447e10e3e4a327e8ac1c598935ddd294ae/src/Http/Http.Extensions/src/ProblemDetailsServiceCollectionExtensions.cs#L22) is called. Previously `_problemDetailsService` would only run if the `_options.ExceptionHandler` is not null. But now that `_problemDetailsService` is part of `IExceptionHandler` it would always run and it will always handle the exception.